DIPs/13NNN-WGB.md

```diff
@@ -428,12 +428,92 @@ Class objects cannot have move constructors or move assignment operators.
 ### Last Use
 
 The *Last Use* of an lvalue is the last time in a path of execution that any read
-or write is done to the memory region referred to directly by the lvalue. For example:
+or write is done to the memory region referred to directly by the lvalue.
 
+**Return Statements**. If an lvalue is returned directly via `return x`, that is the last use of `x`.
+If the return statement is more complex, the rightmost use of the lvalue will be considered the last
```

Comment:
s/is more complex/is an expression using a local `x` several times/

I guess

Reply:
Then please mind really odd cases such as:

```d
return () { ... use x ... }();
```

Reply:
That falls under how references to outer variables in nested functions works.